centreon/src/Core/HostCategory/Infrastructure/Repository/DbReadHostCategoryRepository.php

```diff
@@ -0,0 +1,203 @@
+            FROM `:db`.hostcategories hc
+            INNER JOIN `:db`.acl_resources_hc_relations arhr
+                ON hc.hc_id = arhr.hc_id
+            INNER JOIN `:db`.acl_resources res
+                ON arhr.acl_res_id = res.acl_res_id
+            INNER JOIN `:db`.acl_res_group_relations argr
+                ON res.acl_res_id = argr.acl_res_id
+            INNER JOIN `:db`.acl_groups ag
+                ON argr.acl_group_id = ag.acl_group_id'
+        );
+        $concatenator->storeBindValueMultiple(':access_group_ids', $accessGroupIds, \PDO::PARAM_INT)
+            ->appendWhere('ag.acl_group_id IN (:access_group_ids)');
+
+        return $this->retrieveHostCategories($concatenator, $requestParameters);
+    }
+
+    /**
+     * @param SqlConcatenator $concatenator
+     * @param RequestParametersInterface|null $requestParameters
+     * @return HostCategory[]
+     */
+    private function retrieveHostCategories(
+        SqlConcatenator $concatenator,
+        ?RequestParametersInterface $requestParameters
+    ): array {
+        // Exclude severities from the results
+        $concatenator->appendWhere('hc.level IS NULL');
+
+        // Settup for search, pagination, order
+        $sqlTranslator = $requestParameters ? new SqlRequestParametersTranslator($requestParameters) : null;
+        $sqlTranslator?->setConcordanceArray([
+            'id' => 'hc.hc_id',
+            'name' => 'hc.hc_name',
+            'alias' => 'hc.hc_alias',
+            'is_activated' => 'hc.hc_activate'
+        ]);
+        $sqlTranslator?->addNormalizer('is_activated', new BoolToEnumNormalizer());
+        $sqlTranslator?->translateForConcatenator($concatenator);
+
+        $statement = $this->db->prepare($this->translateDbName($concatenator->__toString()));
+
+        $sqlTranslator?->bindSearchValues($statement);
+        foreach ($concatenator->retrieveBindValues() as $param => [$value, $type]) {
+            $statement->bindValue($param, $value, $type);
+        }
+        $statement->execute();
+
+        $sqlTranslator?->calculateNumberOfRows($this->db);
+
+        $hostCategories = [];
+        while (is_array($result = $statement->fetch(\PDO::FETCH_ASSOC))) {
+            /** @var array{hc_id:int,hc_name:string,hc_alias:string,hc_activate:'0'|'1',hc_comment:string|null} $result */
+            $hostCategories[] = $this->createHostCategoryFromArray($result);
+        }
+
+        return $hostCategories;
+    }
+
+    /**
+     * @param array{hc_id:int,hc_name:string,hc_alias:string,hc_activate:'0'|'1',hc_comment:string|null} $result
+     * @return HostCategory
+     */
+    private function createHostCategoryFromArray(array $result): HostCategory
+    {
+        $hostCategory = new HostCategory(
+            $result['hc_id'],
+            $result['hc_name'],
+            $result['hc_alias']
+        );
+        $hostCategory->setActivated($result['hc_activate'] === self::HC_IS_ACTIVE ? true : false);
```

Comment:
Simple cast to bool ?
```suggestion
        $hostCategory->setActivated((bool) $result['hc_activate']);
```